File: R/generate_dataframe.R
#' Generate a dataframe
#'
#' Generate a dataframe with different types of columns
#'
#' @param num_rows Number of rows to generate
#' @param types Types of columns to generate
#'
#' @return A dataframe with the generated columns
#'
#' @noRd
#'
#' @examples
#' generate_dataframe(10L)
generate_dataframe <- function(num_rows, types = names(vector_generators)) {
  if (rlang::is_empty(types)) {
    return(data.frame()[seq_len(num_rows), ])
  }

  data <- lapply(types, generate_vector, n = num_rows)
  names(data) <- types
  as.data.frame(data)
}

File: inst/scripts/generate_dataframe.py
import pandas as pd
from generate_vector import vector_generators


def generate_dataframe(n_rows, types=None):
    """
    Generate a pandas DataFrame with specified number of rows and column types.

    Parameters:
        n_rows (int): The number of rows in the DataFrame.
        types (list, optional): A list of column types to include in the DataFrame.
                                Choose from the list of vector_generators keys.
                                If not provided, all available column types will be included.

    Returns:
        pandas.DataFrame: The generated DataFrame.

    """
    if types is None:
        types = list(vector_generators.keys())

    data = {t: vector_generators[t](n_rows) for t in types}
    return pd.DataFrame(data)
